core/trino-spi/src/main/java/io/trino/spi/ptf/DescriptorArgument.java

```diff
@@ -50,4 +44,27 @@ public Optional<Descriptor> getDescriptor()
     {
         return descriptor;
     }
+
+    public static Builder descriptorArgumentBuilder()
```

Comment:
why not `builder` (and used without a static import)?

Reply:
Then maybe drop the factory method, and make the Builder's constructor public?

Reply:
Let's keep the factory method and have `DescriptorArgument.builder()`.
IMO it looks significantly nicer than `new DescriptorArguemnt.Builder()`